From 26da33a952f30e0991d8dcc461574adbf7671014 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Fri, 7 Aug 2020 11:30:21 +0530
Subject: [PATCH 31/31] Add user/kernel split Kconfig options for full 1 or 2GB
 RAM.

Very similar to x86, which provides these options for some time already, the
new CONFIG_VMSPLIT_*_OPT options reduce CONFIG_PAGE_OFFSET by 320MB to allow
using all available memory, avoiding messages like this:

Truncating RAM at 10000000-8fffffff to -7f7fffff (vmalloc region overlap).

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/Kconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 8a50efb559f3..a501811363b0 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1315,6 +1315,9 @@ choice
 		bool "3G/1G user/kernel split (for full 1G low memory)"
 	config VMSPLIT_2G
 		bool "2G/2G user/kernel split"
+	config VMSPLIT_2G_OPT
+		depends on !ARM_LPAE
+		bool "2G/2G user/kernel split (for full 2G low memory)"
 	config VMSPLIT_1G
 		bool "1G/3G user/kernel split"
 endchoice
@@ -1323,6 +1326,7 @@ config PAGE_OFFSET
 	hex
 	default PHYS_OFFSET if !MMU
 	default 0x40000000 if VMSPLIT_1G
+	default 0x6C000000 if VMSPLIT_2G_OPT
 	default 0x80000000 if VMSPLIT_2G
 	default 0xB0000000 if VMSPLIT_3G_OPT
 	default 0xC0000000
-- 
2.17.1

